src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -694,24 +703,12 @@ def __call__(
                     if callback is not None and i % callback_steps == 0:
                         callback(i, t, latents)
 
-        if output_type == "latent":
-            image = latents
-            has_nsfw_concept = None
-        elif output_type == "pil":
-            # 8. Post-processing
-            image = self.decode_latents(latents)
-
-            # 9. Run safety checker
-            image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
+        if not output_type == "latent":
+            image = self.vae.decode(latents / self.vae.config.scaling_factor).sample
 
-            # 10. Convert to PIL
-            image = self.numpy_to_pil(image)
-        else:
-            # 8. Post-processing
-            image = self.decode_latents(latents)
+        image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype, output_type=output_type)
```

Comment:
actually differently to what I said  before can we maybe do the following here:

```suggestion
             image, has_nsfw_concept = self.run_safety_checker(image, device, prompt_embeds.dtype)
         else:
             has_nsfw_concept = None
```

Reply:
Let's not run the safety checker if we want to out put `latent` type - sorry this was not a good idea before

Reply:
Also the safety checker here in the pipelines should always return PyTorch format as this is the only format that is accepted by the image processor

Reply:
The safety checker will now in this pipeline always return PyTorch tensors which is what we want